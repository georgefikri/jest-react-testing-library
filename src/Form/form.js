import React, {useState} from 'react'

// create login form /////
// create user name
// create last name
// create submit button 

function Form() {
    const [name, setName] = useState({name:'',lastName:''})

    const handleChange = (event)=> {
        event.persist();
        setName(name => ({...name, [event.target.name]:event.target.value }))
    }

    const handleClick = (e)=> {
      e.preventDefault();
      console.log('name:', name.name)
      console.log('lastname:', name.lastName)
    }

    return (
        <div>
            <form data-testid="form">
                <div>
                    <span>first name</span>
                    <input type='text' data-testid="firstName" placeholder='first name' value={name.name} name='name' onChange={handleChange} />
                </div>
                <div>
                    <span>last name</span>
                    <input type='text' placeholder='last name' value={name.lastName} name='lastName' onChange={handleChange} />
                </div>
                <div>
                    <button data-testid='clickme' onClick={handleClick}>click me</button>
                </div>
            </form>
        </div>
    )
}

export default Form
