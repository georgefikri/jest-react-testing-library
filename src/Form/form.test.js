import React from 'react'
import { render, getByTestId, fireEvent , cleanup } from '@testing-library/react';
import Form from './form'

import renderer from 'react-test-renderer'

describe('test the simple form', ()=> {

    afterEach(cleanup)
   test ('check if id is here', ()=> {
    const { getByTestId } = render(<form data-testid="form" />);
    const linkElement = getByTestId('form')
    expect(linkElement).toBeInTheDocument();
   })

   test('check if firstname exists', ()=> {
       const {getByTestId } = render( <input data-testid="firstName" />);
       const linkElement = getByTestId('firstName')
       expect(linkElement).toBeInTheDocument()
   })

   test('check if submit button exist', ()=> {
       const {getByTestId} = render(<button data-testid='clickme' />)
       const linkElement = getByTestId('clickme')
       expect( getByTestId('clickme')).toBeInTheDocument()
   })

   test('check if submit button is clicked', ()=> {
    const {getByTestId} = render(<button data-testid='clickme' />)
    const linkElement = getByTestId('clickme')
    fireEvent.click(linkElement)

    })

    // snapshot testing
    test('snapshot testing button submit', ()=> {
        const tree = renderer.create( <button data-testid='clickme'>click me</button>).toJSON()
        expect(tree).toMatchSnapshot();
    })
    
    test('snapshot testing the whole form', ()=> {
        const fName = renderer.create(<Form />).toJSON();
        expect(fName).toMatchSnapshot() 
    })
})