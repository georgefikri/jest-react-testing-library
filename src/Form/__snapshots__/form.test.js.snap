// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the simple form snapshot testing button submit 1`] = `
<button
  data-testid="clickme"
>
  click me
</button>
`;

exports[`test the simple form snapshot testing the whole form 1`] = `
<div
  data-testid="formContainer"
>
  <form
    data-testid="form"
  >
    <div
      data-testid="FNCont"
    >
      <span>
        first name
      </span>
      <input
        data-testid="firstName"
        name="name"
        onChange={[Function]}
        placeholder="first name"
        type="text"
        value=""
      />
    </div>
    <div>
      <span>
        last name
      </span>
      <input
        name="lastName"
        onChange={[Function]}
        placeholder="last name"
        type="text"
        value=""
      />
    </div>
    <div>
      <button
        data-testid="clickme"
        onClick={[Function]}
      >
        click me
      </button>
    </div>
  </form>
</div>
`;
